lib/dao.js

```diff
@@ -2696,14 +2709,19 @@ DataAccessObject.updateAll = function(where, data, options, cb) {
           inst = new Model(data, {applyDefaultValues: false});
         }
 
-        // validation required
-        inst.isValid(function(valid) {
-          if (valid) {
+        if (doValidate === false) {
+          doUpdate(where, data);
+        } else {
+          // validation required
+          inst.isValid(function(valid) {
+            if (!valid) {
+              if (doValidate) {
```

Comment:
So if `doValidate` is not true, then we don't need to run `isValid` at all, because any validation errors will be silently discarded anyways.

Considering that some of our built-in validations (like "unique") are pretty expensive, I think we should avoid running `isValid` if it's not necessary.